<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GraphWorks</string>
  </property>
  <property name="windowOpacity">
   <double>1.000000000000000</double>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <property name="dockNestingEnabled">
   <bool>false</bool>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <property name="documentMode">
       <bool>false</bool>
      </property>
      <property name="tabsClosable">
       <bool>false</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
      <property name="tabBarAutoHide">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="graphView">
       <attribute name="title">
        <string>Graph view</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>PushButton</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="adjacencyMatrix">
       <attribute name="title">
        <string>Adjacency matrix</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QPushButton" name="buttonPinAdjMatr">
          <property name="maximumSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="resource.qrc">
            <normaloff>:/ico/icons/Unpin.svg</normaloff>
            <normalon>:/ico/icons/Pin.svg</normalon>:/ico/icons/Unpin.svg</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <spacer name="horizontalSpacer_6">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>511</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="labelNodesCount">
          <property name="text">
           <string>Nodes count</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QSpinBox" name="nodesCountAdjMatr">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Choose nodes count</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QPushButton" name="buttonApplyAdjMatr">
          <property name="toolTip">
           <string>Apply matrix changes</string>
          </property>
          <property name="text">
           <string>Apply</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="5">
         <widget class="QTableWidget" name="tableAdjMatr">
          <row>
           <property name="text">
            <string>Новая строка</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>Новая строка</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>Новая строка</string>
           </property>
          </row>
          <column>
           <property name="text">
            <string>Новый столбец</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Новый столбец</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Новый столбец</string>
           </property>
          </column>
          <item row="0" column="0">
           <property name="text">
            <string>12</string>
           </property>
          </item>
          <item row="0" column="1">
           <property name="text">
            <string>12</string>
           </property>
          </item>
          <item row="0" column="2">
           <property name="text">
            <string>12</string>
           </property>
          </item>
          <item row="1" column="0">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="1" column="1">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="1" column="2">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="2" column="0">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="2" column="1">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="2" column="2">
           <property name="text">
            <string>1</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="edgeList">
       <attribute name="title">
        <string>Edge list</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <widget class="QPushButton" name="buttonPinEdgLst">
          <property name="maximumSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="resource.qrc">
            <normaloff>:/ico/icons/Unpin.svg</normaloff>
            <normalon>:/ico/icons/Pin.svg</normalon>:/ico/icons/Unpin.svg</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <spacer name="horizontalSpacer_5">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>511</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="labelNodesCount_2">
          <property name="text">
           <string>Nodes count</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QSpinBox" name="nodesCountEdgLst">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Choose nodes count</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QPushButton" name="buttonApplyEdgLst">
          <property name="minimumSize">
           <size>
            <width>80</width>
            <height>0</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Apply matrix changes</string>
          </property>
          <property name="text">
           <string>Apply</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="5">
         <widget class="QTableWidget" name="tableEdgLst"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="IncidenceMatrix">
       <attribute name="title">
        <string>Incidence matrix</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="QPushButton" name="buttonPinIncMatr">
          <property name="maximumSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="resource.qrc">
            <normaloff>:/ico/icons/Unpin.svg</normaloff>
            <normalon>:/ico/icons/Pin.svg</normalon>:/ico/icons/Unpin.svg</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>511</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="nodesCount_3">
          <property name="text">
           <string>Nodes count</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QSpinBox" name="nodesCountIncMatr">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Choose nodes count</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QPushButton" name="buttonApplyIncMatr">
          <property name="toolTip">
           <string>Apply matrix changes</string>
          </property>
          <property name="text">
           <string>Apply</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="5">
         <widget class="QTableWidget" name="tableIncMatr">
          <property name="rowCount">
           <number>5</number>
          </property>
          <property name="columnCount">
           <number>6</number>
          </property>
          <row>
           <property name="text">
            <string>Новая строка</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>1</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>2</string>
           </property>
          </row>
          <row/>
          <row/>
          <column>
           <property name="text">
            <string>Новый столбец</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>1</string>
           </property>
          </column>
          <column/>
          <column/>
          <column/>
          <column/>
          <item row="0" column="0">
           <property name="text">
            <string/>
           </property>
          </item>
          <item row="0" column="1">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="0" column="2">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="0" column="3">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="1" column="0">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="1" column="2">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="1" column="3">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="2" column="0">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="2" column="1">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="2" column="3">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="3" column="0">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="3" column="1">
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item row="3" column="2">
           <property name="text">
            <string>1</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
       <zorder>tableIncMatr</zorder>
       <zorder>nodesCountIncMatr</zorder>
       <zorder>horizontalSpacer_4</zorder>
       <zorder>nodesCount_3</zorder>
       <zorder>buttonApplyIncMatr</zorder>
       <zorder>buttonPinIncMatr</zorder>
      </widget>
      <widget class="QWidget" name="textOut">
       <attribute name="title">
        <string>Text output</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="1">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="3">
         <widget class="QPushButton" name="buttonClearConsole">
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>0</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Clear all text outputs</string>
          </property>
          <property name="text">
           <string>Clear</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="buttonPinConsole">
          <property name="maximumSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="resource.qrc">
            <normaloff>:/ico/icons/Unpin.svg</normaloff>
            <normalon>:/ico/icons/Pin.svg</normalon>:/ico/icons/Unpin.svg</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="4">
         <widget class="QPlainTextEdit" name="console">
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="plainText">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="mainMenu">
    <property name="title">
     <string>Main menu</string>
    </property>
    <property name="icon">
     <iconset resource="resource.qrc">
      <normaloff>:/ico/icons/Logo.svg</normaloff>:/ico/icons/Logo.svg</iconset>
    </property>
   </widget>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSave_as"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
   </widget>
   <widget class="QMenu" name="menuAlgorithm">
    <property name="title">
     <string>Algorithm</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuNode_movement">
    <property name="title">
     <string>Node movement</string>
    </property>
    <addaction name="actionManual"/>
    <addaction name="actionAutomatic"/>
   </widget>
   <widget class="QMenu" name="menuView_mode">
    <property name="title">
     <string>View mode</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionUnpin_current_tab"/>
    <addaction name="separator"/>
   </widget>
   <addaction name="mainMenu"/>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuAlgorithm"/>
   <addaction name="menuNode_movement"/>
   <addaction name="menuView_mode"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolTip">
    <string/>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionUndo"/>
   <addaction name="actionRedo"/>
   <addaction name="actionLaunchAlg"/>
   <addaction name="actionDeleteGraph"/>
   <addaction name="actionAddNode"/>
   <addaction name="actionDeleteNode"/>
  </widget>
  <action name="actionUndo">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/ico/icons/Undo.svg</normaloff>:/ico/icons/Undo.svg</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/ico/icons/Redo.svg</normaloff>:/ico/icons/Redo.svg</iconset>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/ico/icons/Save.svg</normaloff>:/ico/icons/Save.svg</iconset>
   </property>
   <property name="text">
    <string>Save as</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionAutomatic">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/ico/icons/autoMov.svg</normaloff>:/ico/icons/autoMov.svg</iconset>
   </property>
   <property name="text">
    <string>Automatic</string>
   </property>
  </action>
  <action name="actionManual">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/ico/icons/manualMov.svg</normaloff>:/ico/icons/manualMov.svg</iconset>
   </property>
   <property name="text">
    <string>Manual</string>
   </property>
  </action>
  <action name="actionLaunchAlg">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/ico/icons/launchAlg.svg</normaloff>:/ico/icons/launchAlg.svg</iconset>
   </property>
   <property name="text">
    <string>LaunchAlg</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actionDeleteGraph">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/ico/icons/Clear.svg</normaloff>:/ico/icons/Clear.svg</iconset>
   </property>
   <property name="text">
    <string>DeleteGraph</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actionAddNode">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/ico/icons/addNode.svg</normaloff>:/ico/icons/addNode.svg</iconset>
   </property>
   <property name="text">
    <string>AddNode</string>
   </property>
   <property name="toolTip">
    <string>Add new node</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actionDeleteNode">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/ico/icons/deleteNode.svg</normaloff>:/ico/icons/deleteNode.svg</iconset>
   </property>
   <property name="text">
    <string>DeleteNode</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actionClearConsole">
   <property name="text">
    <string>ClearConsole</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actionUnpin_current_tab">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/ico/icons/Unpin.svg</normaloff>:/ico/icons/Unpin.svg</iconset>
   </property>
   <property name="text">
    <string>Unpin current tab</string>
   </property>
  </action>
  <action name="actionShown_tabs">
   <property name="text">
    <string>Shown tabs</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections/>
</ui>
